app/controllers/katello/api/v2/content_credentials_controller.rb

```diff
@@ -0,0 +1,115 @@
+module Katello
+  class Api::V2::ContentCredentialsController < Api::V2::ApiController
+    include Katello::Concerns::FilteredAutoCompleteSearch
+    before_action :authorize
+    before_action :find_organization, :only => [:create, :index, :auto_complete_search]
+    before_action :find_content_credential, :only => [:show, :update, :destroy, :content, :set_content]
+    skip_before_action :check_content_type, :only => [:create, :content, :set_content]
+
+    def resource_class
+      Katello::GpgKey
+    end
+
+    def_param_group :content_credential do
+      param :name, :identifier, :action_aware => true, :required => true, :desc => N_("identifier of the content credential")
+      param :content_type, String, :action_aware => true, :required => true, :desc => N_("type of content")
+      param :content, String, :action_aware => true, :required => true, :desc => N_("public key block in DER encoding / certificate content")
+    end
+
+    resource_description do
+      description <<-DESC
+        # Description
+
+        Content Credentials are used to store credentials like GPG Keys and Certificates for the authentication
+        to Products / Repositories.
+      DESC
+      api_version "v2"
+    end
+
+    api :GET, "/content_credentials", N_("List content credentials")
+    param :organization_id, :number, :desc => N_("organization identifier"), :required => true
+    param :name, String, :desc => N_("name of the Content Credential"), :required => false
+    param :content_type, String, :desc => N_("type of content"), :required => false
+    param_group :search, Api::V2::ApiController
+    def index
+      respond(:collection => scoped_search(index_relation.distinct, :name, :asc))
+    end
+
+    def index_relation
+      query = GpgKey.readable.where(:organization_id => @organization.id)
+      query = query.where(:name => params[:name]) if params[:name]
+      query
+    end
```

Comment:
I don't actually see the content_type param being used here, shouldn't it be treated similarly to name?